bitwarden_license/src/Sso/Utilities/Saml2OptionsExtensions.cs

```diff
@@ -81,13 +81,24 @@ public static async Task<bool> CouldHandleAsync(this Saml2Options options, strin
                 return false;
             }
 
-            if (assertion == null)
+            if (envelope == null)
             {
                 return false;
             }
 
+            if (options.SPOptions.WantAssertionsSigned)
+            {
+                var assertion = envelope["Assertion", Saml2Namespaces.Saml2Name];
+                var IsAssertionSigned = assertion != null && XmlHelpers.IsSignedByAny(assertion, idp.SigningKeys,
+                    options.SPOptions.ValidateCertificates, options.SPOptions.MinIncomingSigningAlgorithm);
+                if (!IsAssertionSigned)
+                {
+                    throw new Exception("Cannot verify SAML assertion signature.");
+                }
+            }
+
             // Double check the entity Ids
-            var entityId = assertion["Issuer", Saml2Namespaces.Saml2Name]?.InnerText.Trim();
+            var entityId = envelope["Issuer", Saml2Namespaces.Saml2Name]?.InnerText.Trim();
             return string.Equals(entityId, idp.EntityId.Id, StringComparison.InvariantCultureIgnoreCase);
```

Comment:
Basically, instead of returning here, make this check, and then if `false`; `return false;`; then you can fall through to checking the signature configuration, and `return true;` at the end of the method if no exception has been thrown.